File: app/src/main/java/com/dhu/dhu_timetable/repo/TimetableRepo.java
package com.dhu.dhu_timetable.repo;

import android.util.Log;

import androidx.lifecycle.LiveData;

import com.dhu.dhu_timetable.service.APIService;
import com.dhu.dhu_timetable.service.RetrofitConnect;
import com.dhu.dhu_timetable.network.TimetableApiClient;
import com.dhu.dhu_timetable.model.TimetableModel;

import java.util.List;

public class TimetableRepo {
    // 싱글톤 패턴
    private static TimetableRepo instance;

    private TimetableApiClient mTimetableApiClient;

    public static TimetableRepo getInstance() {
        if (instance == null)
            instance = new TimetableRepo();
        return instance;
    }
    private APIService service;
    String TAG = "janghee";

    public TimetableRepo() {
        mTimetableApiClient = TimetableApiClient.getInstance();
        service = RetrofitConnect.getRetrofitClient().create(APIService.class);
        Log.d(TAG, "TimetableRepo: 성공");
    }

    public LiveData<List<TimetableModel>> getTimetable() {
        return mTimetableApiClient.getTimetable();
    }

    // 3- Calling the method in repository
    public void setTimetableApi(String email) {
        mTimetableApiClient.getTimetableData(email);

    }

    public void insertTimetableApi(String email, String subjectName, String workDay, String cyber, String quarter) {
        mTimetableApiClient.setTimetableData(email, subjectName, workDay, cyber, quarter);
    }

    public void nextInsertTimetableApi(String email) {
        setTimetableApi(email);
    }

    public void deleteTimetableApi(String email, int id) {
        mTimetableApiClient.deleteTimetableData(email, id);
    }

}

File: app/src/main/java/com/dhu/dhu_timetable/repo/TimetableRepo.kt
package com.dhu.dhu_timetable.repo

import androidx.lifecycle.LiveData
import com.dhu.dhu_timetable.model.TimetableModel
import com.dhu.dhu_timetable.network.TimetableApiClient

class TimetableRepo {

    private val mTimetableApiClient: TimetableApiClient by lazy { TimetableApiClient.getInstance() }

    companion object {
        @Volatile private var instance: TimetableRepo? = null
        @JvmStatic
        fun getInstance(): TimetableRepo =
                instance ?: synchronized(this) {
                    instance ?: TimetableRepo().also { instance = it }
                }
    }

    fun getTimetable(): LiveData<List<TimetableModel>> {
        return mTimetableApiClient.timetable
    }

    fun setTimetableApi(email: String) {
        mTimetableApiClient.getTimetableData(email)
    }

    fun insertTimetableApi(email: String, subjectName: String, workDay: String, cyber: String, quarter: String) {
        mTimetableApiClient.setTimetableData(email, subjectName, workDay, cyber, quarter)
    }

    fun nextInsertTimetableApi(email: String) {
        setTimetableApi(email)
    }

    fun deleteTimetableApi(email: String, id: Int) {
        mTimetableApiClient.deleteTimetableData(email, id)
    }

}